frame/support/procedural/src/construct_runtime/parse.rs

```diff
@@ -198,6 +199,84 @@ impl Parse for PalletDeclaration {
 	}
 }
 
+/// A struct representing a path to a pallet. `PalletPath` is almost identical to the standard
+/// Rust path with a few restrictions:
+/// - No leading colons allowed
+/// - Path segments can only consist of identifers; angle-bracketed or parenthesized segments will
+///   result in a parsing error (except when specifying instances)
+#[derive(Debug, Clone)]
+pub struct PalletPath {
+	pub inner: Path,
+}
+
+impl Parse for PalletPath {
+	fn parse(input: ParseStream) -> Result<Self> {
+		let mut lookahead = input.lookahead1();
+		let mut segments = Punctuated::new();
+
+		if lookahead.peek(Token![crate])
+			|| lookahead.peek(Token![self])
+			|| lookahead.peek(Token![super])
+			|| lookahead.peek(Ident)
+		{
+			let ident = input.call(Ident::parse_any)?;
+			segments.push(PathSegment { ident, arguments: PathArguments::None });
+			let _: Token![::] = input.parse()?;
+			lookahead = input.lookahead1();
+		} else {
+			return Err(lookahead.error());
+		}
+
+		while lookahead.peek(Ident) {
+			let ident = input.parse()?;
+			segments.push(PathSegment { ident, arguments: PathArguments::None });
+			let _: Token![::] = input.parse()?;
+			lookahead = input.lookahead1();
+		}
+
+		if !lookahead.peek(token::Brace) && !lookahead.peek(Token![<]) {
+			return Err(lookahead.error());
+		}
+
+		Ok(Self {
+			inner: Path {
+				leading_colon: None,
+				segments,
+			}
+		})
+	}
+}
+
+impl PalletPath {
+	/// Return the snake-cased module name for this path.
+	pub fn mod_name(&self) -> Ident {
+		let mut iter = self.inner.segments.iter();
+		let mut mod_name = match &iter.next().expect("Path should always have 1 segment; qed").ident {
+			ident if ident == "self" || ident == "super" || ident == "crate" => {
+				// Skip `crate`, `self` and `super` quasi-keywords when creating the module name
+				iter.next()
+					.expect("There must be a path segment pointing to a pallet following \
+						`crate`, `self` or `super`; qed")
```

Comment:
in the future I would like to be able to use simply `crate` for the path of the pallet in tests for example.
But this method should be removed once we start using the pallet name for field/variant names instead of this path.
So it is ok.

But maybe we can still support them by prefixing with `_` or something